sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -452,6 +452,45 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     // scalastyle:on
   }
 
+  test("overlay for byte array") {
+    val input = Literal(Array[Byte](1, 2, 3, 4, 5, 6, 7, 8, 9))
+    checkEvaluation(new Overlay(input, Literal(Array[Byte](-1)),
+      Literal.create(6, IntegerType)), Array[Byte](1, 2, 3, 4, 5, -1, 7, 8, 9))
```

Comment:
Nit: Also test corner case, e.g. position is negative, or position is larger than 9

Reply:
Thanks for your reminder. but the negative is illegal in `Postgresql`.
```
org.postgresql.util.PSQLException: ERROR: negative substring length not allowed
  在位置：SQL function "overlay" statement 1
```
I will check position is larger than 9.

Reply:
I will add tests about your suggestion in `StringExpressionsSuite`.

Reply:
I think Spark basically avoids runtime exceptions. Could you follow the behaivour for string cases?
```
scala> sql("""select overlay("aaaaaa", "bb", 0)""").show
+--------------------------+                                                    
|overlay(aaaaaa, bb, 0, -1)|
+--------------------------+
|                   bbaaaaa|
+--------------------------+


scala> sql("""select overlay("aaaaaa", "bb", -10)""").show
+----------------------------+
|overlay(aaaaaa, bb, -10, -1)|
+----------------------------+
|                    bbaaaaaa|
+----------------------------+
```

Reply:
I have added tests for negative and zero. I don't understand this.

Reply:
In this pr, plz follow the current behaivour of overlay(string, string). If you wanna to change this behaivour, you can make an another pr for that.

Reply:
@maropu OK. I will follow the current behaivour of overlay(string, string).